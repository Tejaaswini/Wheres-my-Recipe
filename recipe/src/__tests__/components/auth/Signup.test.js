import React from 'react';
import ReactDOM from 'react-dom';
import configureStore from 'redux-mock-store';
import { connect } from 'react-redux';
import { Provider } from 'react-redux';
import {mount, shallow} from 'enzyme';
import { Field, reduxForm } from 'redux-form';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Button, Form, FormGroup, FormFeedback, Label, Input } from 'reactstrap';
import CarouselComponent from '../../../components/universal/carousel';
import Signup from '../../../components/auth/signup';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('Signup', () => {
    let store, wrapper;
    let initialState = {
        auth: {
            authenticated: false
        },
        form: {},
        categories: {},
        category: {},
        recipes: {},
        recipe: {},
    };

    beforeEach(() => {
        store = mockStore(initialState);
    });
    it('it renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Provider store={store}>
                <Router>
                    <Signup />
                </Router>
            </Provider>, div
        );
    });

    it('it has valid snapshot', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router>
                    <Signup />
                </Router>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('it renders <CarouselComponent /> component', () => {
        const wrapper = mount((
            <Provider store={store}>
                <Router>
                    <Signup />
                </Router>
            </Provider>
        ));
        expect(wrapper.find(CarouselComponent)).toHaveLength(1);
    });

    it('it renders <CarouselComponent /> component', () => {
        const wrapper = mount((
            <Provider store={store}>
                <Router>
                    <Signup />
                </Router>
            </Provider>
        ));
        expect(wrapper.find(CarouselComponent)).toHaveLength(1);
    });

    it('it renders four <Field /> component', () => {
        const wrapper = mount((
            <Provider store={store}>
                <Router>
                    <Signup />
                </Router>
            </Provider>
        ));
        expect(wrapper.contains(Field)).toEqual(true);
        expect(wrapper.find(Field)).toHaveLength(4);
    });
});