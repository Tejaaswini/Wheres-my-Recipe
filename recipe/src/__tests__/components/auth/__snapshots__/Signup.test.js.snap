// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup it has valid snapshot 1`] = `
<div>
  <div
    className="carousel slide"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        onClick={[Function]}
      />
      <li
        className=""
        onClick={[Function]}
      />
      <li
        className=""
        onClick={[Function]}
      />
    </ol>
    <div
      className="carousel-inner"
      role="listbox"
    >
      <div
        className="carousel-item active"
      >
        <img
          alt={undefined}
          className="d-block img-fluid"
          src={undefined}
        />
        <div
          className="carouselimg"
        >
          <img
            alt="Slide 1"
            src="recipe-food-lab.jpg"
          />
        </div>
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3>
            Slide 1
          </h3>
          <p>
            Slide 1
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt={undefined}
          className="d-block img-fluid"
          src={undefined}
        />
        <div
          className="carouselimg"
        >
          <img
            alt="Slide 2"
            src="recipe-kenji-14.jpg"
          />
        </div>
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3>
            Slide 2
          </h3>
          <p>
            Slide 2
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt={undefined}
          className="d-block img-fluid"
          src={undefined}
        />
        <div
          className="carouselimg"
        >
          <img
            alt="Slide 3"
            src="recipes-roundup-22.jpg"
          />
        </div>
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3>
            Slide 3
          </h3>
          <p>
            Slide 3
          </p>
        </div>
      </div>
    </div>
    <a
      className="carousel-control-prev"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>
  <div
    className="container"
  >
    <div
      className="row justify-content-md-center"
    >
      <div
        className="col-md-6"
      >
        <h3>
          CREATE ACCOUNT
        </h3>
        <hr />
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className=" form-control-label"
              htmlFor="firstname"
            >
              Firstname
            </label>
            <input
              className="is-valid form-control"
              id="firstname"
              name="first_name"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className=" form-control-label"
              htmlFor="lastname"
            >
              Lastname
            </label>
            <input
              className="is-valid form-control"
              id="lastname"
              name="last_name"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className=" form-control-label"
              htmlFor="email"
            >
              Email
            </label>
            <input
              className="is-valid form-control"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="email"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className=" form-control-label"
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="is-valid form-control"
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="password"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              
            </div>
          </div>
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="submit"
          >
            Sign up
          </button>
        </form>
        <a
          href="/signin"
          onClick={[Function]}
        >
          ALREADY A MEMBER! SIGN IN 
          <i
            className="fa fa-arrow-right"
          />
        </a>
      </div>
    </div>
    <hr
      className="featurette-divider"
    />
  </div>
  <div
    id="notification-wrapper"
  />
</div>
`;
